src/others/helpers/map-utils.ts

```diff
@@ -50,3 +27,34 @@ export const mapAidRequestsToFeatures = (mapData: RequestMapDataPoint[], locatio
   });
 };
 
+export type RegionMetadata = { [id: string]: {amount: number, description: string } };
+type RegionData = {
+  region_id: string;
+  city_name: string;
+  amount: number;
+};
+
+export const mapRegionIdsToMetadata = (mapDataPoints: MapDataPoint[], translateLocation: (city_id: number) => Location | undefined): RegionMetadata => {
+  const regionData: RegionData[] = mapDataPoints.map((req) => {
+    const city = translateLocation(req.city_id);
+    if (!city) throw new Error(`Loccation ${req.city_id} is not found`);
+    return {
+      region_id: city.region_id,
+      city_name: city.name,
+      amount: req.amount,
+    };
+  });
+  const groupedRegionData = Object.entries(groupBy(regionData, "region_id"));
+  const regionToMetadata: RegionMetadata = {};
+  groupedRegionData.forEach(([region_id, data]) => {
```

Comment:
**issue**
I don't really understand, why use `forEach` if we could easily use 2 lodash functions and make this code much clearer.

Reply:
I really don't believe you're a noob :D

It's not about styleguides, the way I see it at least. It's about that you listed 3 things that are happening there at once, while in an ideal world I'd see one at a time. 

The only style-related issue is that you're mutating data, which we aim at avoiding, that's why we use map/filter/reduce and all the derivatives instead.

Reply:
TypeScript - noob.

Done.
Found a way how to create a dict with a map, so changed forEach with a map and now the inside computation does not change anything outside its scope.